### ISSUE TITLE
GLideN64 settings window needs "Apply" button

### ISSUE THREAD

**Issue by MofoMan2000** [gonetz/GLideN64#486]
It's really great to be able to fiddle with settings, and see the effect without closing the settings window (by clicking "OK"). There ought to be a button to apply settings without closing the dialog, or changes should take effect immediately.

**Comment by thecoreyburton** [comment]
I think the problem is that for the settings to take effect the emulator needs to render the next frame (meaning the game needs to be playing) and opening plugin settings stops emulation temporarily to allow you to alter these settings. Certain texture filtering settings don't apply when the plugin dialogue is closed and require emulation to be ended and restarted too, as is with the nature of N64 emulation.

**Comment by AmbientMalice** [comment]
@thecoreyburton In PJ64, pausing emulation on window focus lost is optional behavior which can be changed from the settings menu. A lot of people turn it off so games will keep running while they do other things.

**Comment by thecoreyburton** [comment]
My bad, I don't really use PJ64. But in that case, an instant update would definitely be nice, especially when you're tweaking settings and trying to get everything right and don't want to keep having to open and close dialogue boxes. I'd love to see this feature too!

I know this is slightly deviating from the topic, but maybe marking the settings that require emulation restart to take affect properly with an asterisk or something would be useful too if this feature was implemented, so users knew what settings they'd need to restart emulation in order to take effect properly.

**Comment by MofoMan2000** [comment]
@AmbientMalice Exactly, that's also why I think it would be great if the video could go to fullscreen on one monitor and not exit fullscreen when the emulator loses focus (like you're doing stuff on the other monitor). Dolphin doesn't exit fullscreen when you're doing other stuff, but every N64 video plugin I've tried does.

I'm trying to file issues which may help the overall usability and interface of the plugin, rather than all these random game-specific issues that clutter up the issue tracker. I'm afraid my issues are getting lost in the deluge though.

**Comment by gonetz** [comment]
Apply button is useful, but hard to implement feature. GLideN64 User Interface is not designed for that.

If my new campaign will be successful
https://www.indiegogo.com/projects/gliden64-fx/x/8367908
I'll have to find a way to implement Apply functionality.

**Comment by MofoMan2000** [comment]
That's cool, hope your new funding campaign is successful. If you get to it, it'll be great, it's a great usability feature and time saver.

**Comment by gonetz** [comment]
I've tried to implement Apply button functionality and met unexpected problem.
Settings change usually means that GL context and all GL objects need to be re-created with new settings. It happens when you press OK button - dialog closed, context updated. 'Apply' button works almost the same as 'Ok' one, but dialog is not closed.

Plugin calls video().restart() method to create new graphics context. OGLVideo::restart() calls OGLVideoWindows::_resizeWindow(), which freezes on Win API SetWindowPos call if config dialog is still open. When I close the dialog, plugin continue to run as usual. I'm not Win API expert and have no idea, why that happens. I did not find a workaround for the problem. Help needed.

**Comment by oddMLan** [comment]
I second this. it would be incredibly useful. Even if the dialog has to be closed, because sometimes I don't want to save the settings to my custom ini

**Comment by gonetz** [comment]
I failed to make it in 2015. I still need help with that task.
